A66-otel-stats.md

```diff
@@ -0,0 +1,468 @@
+  // If `SetMeterProvider()` is not called, the stats plugin is a no-op.
+  OpenTelemetryPluginBuilder& SetMeterProvider(
+      std::shared_ptr<opentelemetry::metrics::MeterProvider> meter_provider);
+  // Enable metrics in \a metric_names. Only these metrics are recorded by gRPC.
+  // Sample -
+  // OpenTelemetryPluginBuilder().EnableMetrics(BaseMetrics()).SetMeterProvider(mp).BuildAndRegisterGlobal();
+  OpenTelemetryPluginBuilder& EnableMetrics(
+      const absl::flat_hash_set<absl::string_view>& metric_names);
+  // The base set of metrics -
+  // grpc.client.attempt.started
+  // grpc.client.attempt.duration
+  // grpc.client.attempt.sent_total_compressed_message_size
+  // grpc.client.attempt.rcvd_total_compressed_message_size
+  // grpc.server.call.started
+  // grpc.server.call.duration
+  // grpc.server.call.sent_total_compressed_message_size
+  // grpc.server.call.rcvd_total_compressed_message_size
+  static absl::flat_hash_set<std::string> BaseMetrics();
+  // If \a generic_method_filter returns true for a method_name, that method_name is recorded as is, otherwise it is recorded as "other".
+  OpenTelemetryPluginBuilder&  SetGenericMethodFilter(absl::AnyInvocable<bool(absl::string_view /*method_name*/)> generic_method_filter);
+  // If \a target_filter returns true for a target, that target is recorded as is, otherwise it is recorded as "other".
+  OpenTelemetryPluginBuilder&  SetTargetFilter(absl::AnyInvocable<bool(absl::string_view /*target*/)> target_filter);
+  // Builds and registers a OpenTelemetry Plugin
+  void BuildAndRegisterGlobal();
+};
+
+```
+
+In the future, additional API might be provided to allow registering the plugin
+for a particular channel or server builder.
+
+### Java
+
+```
+public static class OpenTelemetryModuleBuilder {
+    /**
+     * OpenTelemetry instance is used to configure metrics settings.
+     *
+     * Sample
+     *    SdkMeterProvider sdkMeterProvider = SdkMeterProvider.builder()
+     *         .registerMetricReader(
+     *             PeriodicMetricReader.builder(
+     *                OtlpGrpcMetricExporter.builder().build()).build())
+     *         .build();
+     *
+     *     OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+     *         .setMeterProvider(sdkMeterProvider)
+     *         .build();
+     *
+     * If MeterProvider is not configured, no-op meterProvider will be used by default.
+     * It provides meters which do not record or emit.
+     */
+    public OpenTelemetryModuleBuilder openTelemetry(OpenTelemetry openTelemetry);
+
+    /* Enable metrics for listed metrics. */
+    public OpenTelmetryBuilder enableMetrics(Set<String> metricNames);
+
+    /* If targetFilter returns true for a target, target is recorded as is.
+     * Otherwise it will be recorded as "other". */
+    public OpenTelemetryBuilder targetFilter(Predicate<String> targetFilter);
+
+    public OpenTelemetryModule build();
+}
+```
+
+### Go
+
+To be filled
+
+### Python
+
+To be filled
```

Comment:
@XuanWang-Amos Do you have some idea of what the python API for this will look like?  If not, we can add this later.

Reply:
Not yet, I need discuss with Richard first, this can be added later.

Reply:
Added Python API